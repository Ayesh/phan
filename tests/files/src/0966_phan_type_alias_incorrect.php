<?php

class A {
}

/**
 * @phan-type false = true
 * @phan-type true = false
 * @phan-type void = A
 * @phan-type B = A
 * @phan-type C = B
 * @phan-type B = C
 * @phan-type D =
 * @phan-type E
 * @param C $x
 * @param B $y
 */
function example($x, $y): void {
    '@phan-debug-var $x, $y';
}
example(new A(), new A());
