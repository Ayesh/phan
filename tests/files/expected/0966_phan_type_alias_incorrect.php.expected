%s:7 PhanTypeAliasInternalTypeConflict Saw attempt to use false as a type alias for true but internal types cannot be redefined.
%s:8 PhanTypeAliasInternalTypeConflict Saw attempt to use true as a type alias for false but internal types cannot be redefined.
%s:9 PhanTypeAliasInternalTypeConflict Saw attempt to use void as a type alias for A but internal types cannot be redefined.
%s:13 PhanCommentUnextractableTypeAlias Saw a line * @phan-type D = with a type alias that could not be extracted
%s:14 PhanCommentUnextractableTypeAlias Saw a line * @phan-type E with a type alias that could not be extracted
%s:18 PhanDebugAnnotation @phan-debug-var requested for variable $x - it has union type \A
%s:18 PhanDebugAnnotation @phan-debug-var requested for variable $y - it has union type \A
